src/unitquaternion.jl

```diff
@@ -20,17 +20,17 @@ struct QuatRotation{T} <: Rotation{3,T}
     @inline function QuatRotation{T}(w, x, y, z, normalize::Bool = true) where T
         if normalize
             inorm = inv(sqrt(w*w + x*x + y*y + z*z))
-            new{T}(Quaternion(w*inorm, x*inorm, y*inorm, z*inorm, true))
+            new{T}(Quaternion(w*inorm, x*inorm, y*inorm, z*inorm))
         else
-            new{T}(Quaternion(w, x, y, z, true))
+            new{T}(Quaternion(w, x, y, z))
         end
     end
 
-    @inline function QuatRotation{T}(q::Quaternion) where T
-        if q.norm
-            new{T}(q)
+    @inline function QuatRotation{T}(q::Quaternion, normalize::Bool = true) where T
```

Comment:
This appears to be missing a check for `iszero(q)` (on either the input or output - undecided), which should probably result in an error. I'd be willing to permit `!isfinite(q)` to pass (the result will presumably be a `NaN` quaternion) but someone may prefer that errors as well.

Reply:
The current behavior of constructors other than the `Quaternion` is also not checked for input correctness, so I would like to separate the feature into another PR.
```julia
julia> AngleAxis(0.0, 0.0, 0.0, 0.0)
3×3 AngleAxis{Float64} with indices SOneTo(3)×SOneTo(3)(0.0, NaN, NaN, NaN):
 NaN  NaN  NaN
 NaN  NaN  NaN
 NaN  NaN  NaN

julia> QuatRotation(0.0, 0.0, 0.0, 0.0)
3×3 QuatRotation{Float64} with indices SOneTo(3)×SOneTo(3)(QuaternionF64(0.0, 0.0, 0.0, 0.0)):
 0.0  0.0  0.0
 0.0  0.0  0.0
 0.0  0.0  0.0

julia> RotMatrix{3}([1 1 1;2 2 2;3 3 3])
3×3 RotMatrix3{Int64} with indices SOneTo(3)×SOneTo(3):
 1  1  1
 2  2  2
 3  3  3
```